nevergrad/optimization/metamodel.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import numpy as np
+import nevergrad.common.typing as tp
+from . import utils
+from .base import registry
+from . import callbacks
+
+
+class MetaModelFailure(ValueError):
+    """Sometimes the optimum of the metamodel is at infinity."""
+
+
+def _learn_on_k_best(archive: utils.Archive[utils.MultiValue], k: int) -> tp.ArrayLike:
```

Comment:
make it non-private since we use it in other files
```suggestion
def learn_on_k_best(archive: utils.Archive[utils.MultiValue], k: int) -> tp.ArrayLike:
```

Reply:
Fixed.